examples/java8/src/main/java/org/apache/beam/examples/complete/game/AggregatedTeamScore.java

```diff
@@ -0,0 +1,183 @@
+  configureCompleteWindowedTableWrite() {
+
+    Map<String, WriteWindowedToBigQuery.FieldInfo<KV<String, Integer>>> tableConfigure =
+        new HashMap<String, FieldInfo<KV<String, Integer>>>();
+    tableConfigure.put(
+        "team",
+        new WriteWindowedToBigQuery.FieldInfo<KV<String, Integer>>(
+            "STRING", (c, w) -> c.element().getKey()));
+    tableConfigure.put(
+        "total_score",
+        new WriteWindowedToBigQuery.FieldInfo<KV<String, Integer>>(
+            "INTEGER", (c, w) -> c.element().getValue()));
+    tableConfigure.put(
+        "processing_time",
+        new WriteWindowedToBigQuery.FieldInfo<KV<String, Integer>>(
+            "STRING", (c, w) -> fmt.print(Instant.now())));
+    return tableConfigure;
+  }
+
+
+  public static void main(String[] args) throws Exception {
+
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+    // Enforce that this pipeline is always run in streaming mode.
+    options.setStreaming(true);
+    ExampleUtils exampleUtils = new ExampleUtils(options);
+    Pipeline pipeline = Pipeline.create(options);
+
+    pipeline
+        // Read game events from Pub/Sub using custom timestamps, which are extracted from the
+        // pubsub data elements, and parse the data.
+        .apply(PubsubIO.readStrings()
+            .withTimestampAttribute(TIMESTAMP_ATTRIBUTE).fromTopic(options.getTopic()))
+        .apply("ParseGameEvent", ParDo.of(new ParseEventFn()))
+        // Create <team, GameActionInfo> mapping. UpdateTeamScore uses team name as key.
+        .apply("MapTeamAsKey", MapElements
+            .into(TypeDescriptors.kvs(TypeDescriptors.strings(),
+                TypeDescriptor.of(GameActionInfo.class)))
+            .via((GameActionInfo gInfo) -> KV.of(gInfo.team, gInfo)))
+        // Outputs team score every time a team reaches the threshold.
+        .apply("UpdateTeamScore",
+            ParDo.of(new UpdateTeamScoreFn(options.getThreshold())))
+        // Write the results to BigQuery.
+        .apply(
+            "WriteTeamLeaders",
+            new WriteWindowedToBigQuery<KV<String, Integer>>(
+                options.as(GcpOptions.class).getProject(),
+                options.getDataset(),
+                options.getLeaderBoardTableName() + "_team_leader",
+                configureCompleteWindowedTableWrite()));
+
+    // Run the pipeline and wait for the pipeline to finish; capture cancellation requests from the
+    // command line.
+    PipelineResult result = pipeline.run();
+    exampleUtils.waitToFinish(result);
+  }
+
+  // Outputs team score every time a team reaches the threshold.
+  @VisibleForTesting
+  static class UpdateTeamScoreFn
+      extends DoFn<KV<String, GameActionInfo>, KV<String, Integer>> {
+
+    private final int threshold;
+
+    public UpdateTeamScoreFn(int threshold) {
+      this.threshold = threshold;
+    }
+
+    @StateId("totalScore")
```

Comment:
move the reused strings into a constant?

Reply:
done